### DISCUSSION TITLE
Compiler warning in keylayouts.h line 169

### DISCUSSION THREAD

**Discussion by kmanc** [ArminJo/DigistumpArduino#54] [General]
The [KEY_ENTER](https://github.com/ArminJo/DigistumpArduino/blob/master/digistump-avr/libraries/DigisparkKeyboard/keylayouts.h#L168) definition throws a compiler warning due to integer truncation. This warning can be fixed by removing the `| 0xF000` part of that line, but since I'm a little unclear on what that actually does I'm hesitant to do so. In local testing the key still works if I remove that OR, but I assume it is there for a reason that I just don't understand. Could you explain why that is there?

**Comment by ArminJo** [comment]
I am sorry, but I am also not clear what the OR is for. I belive it is for functions not implemented in this core, but I am not sure. It is a source provided by PJRC.
